iamtest/mockpermissiontester.go

```diff
@@ -0,0 +1,84 @@
+package iamtest
+
+import (
+	"context"
+	"sync"
+
+	"go.einride.tech/aip/resourcename"
+	"go.einride.tech/iam/iamresource"
+
+	iamv1 "go.einride.tech/iam/proto/gen/einride/iam/v1"
+)
+
+// MockPermissionTester is a mock permission tester.
+type MockPermissionTester struct {
+	mu                 sync.Mutex
+	allowAll           bool
+	allowedPermissions []allowedPermission
+}
+
+// NewMockPermissionTester creates a new mock permission tester.
+func NewMockPermissionTester() *MockPermissionTester {
```

Comment:
Since the package is called `iamtest` I think this should be called `NewPermissionTester`.

Reply:
Good point! Fixing

Reply:
Updated!